code/game/objects/items/devices/radio/headset.dm

```diff
@@ -259,15 +267,9 @@
 		if(src == user.wear_ear) //worn
 			var/datum/mob_hud/H = huds[MOB_HUD_SQUAD]
 			if(headset_hud_on)
-				H.add_hud_to(usr)
-				if(user.mind && user.assigned_squad && user.hud_used && user.hud_used.locate_leader)
-					user.hud_used.locate_leader.alpha = 255
-					user.hud_used.locate_leader.mouse_opacity = 1
+				add_squadhud(H, user)
```

Comment:
This should only be added for mobs with assigned squads. Else the doctors, SOs and whatever will be calling this proc all the time, wasting resources.

Reply:
Though ideally there's a separate verb to toggle the tracker as well.

Reply:
The `else` will now prevent from shipside roles from having a squad HUD.
Make these two different checks, please, and different verbs to toggle.